src/infer_stats.cc

```diff
@@ -48,7 +60,8 @@ InferenceStatsAggregator::UpdateFailure(
 
 #ifdef TRITON_ENABLE_METRICS
   if (metric_reporter != nullptr) {
-    metric_reporter->IncrementCounter("inf_failure", 1);
+    std::string reason_str = failureReasonToString(reason);
+    metric_reporter->IncrementCounter("inf_failure_" + reason_str, 1);
```

Comment:
Didn't we agree to use prometheus `labels` for these metrics instead of embedding the reason within the metric name?